### PR TITLE
nautilus: mon: check probe against quorum_features instead

### PR DESCRIPTION
**Pull request by jecluis**
backport of https://github.com/ceph/ceph/pull/28671 which is still open (!)

---

mon: check probe against quorum_features instead

Fixes: http://tracker.ceph.com/issues/40081

Signed-off-by: Joao Eduardo Luis <joao@suse.com>
(cherry picked from commit 8e54c895e8c9a26da2ca8ad76bf308eb125b8013)

Conflicts:
      - mon/Monitor.cc: due to previous selective cherry-pick 9d0404b0350,
      in which we ignored a few recent changes; those would be hard to
      backport due to them overhauling feature naming and access (i.e.,
      ceph_release_t). As such, we are, once more, selectively patching to
      ignore those changes.

also adds a commit to allow the monitor to shutdown gracefully if required features are not met, instead of just probing forever.

### CODE REVIEW

File: src/mon/Monitor.cc [hunk 1]
```diff
@@ -1849,13 +1864,23 @@ void Monitor::handle_probe_probe(MonOpRequestRef op)
 
   dout(10) << "handle_probe_probe " << m->get_source_inst() << *m
 	   << " features " << m->get_connection()->get_features() << dendl;
-  uint64_t missing = required_features & ~m->get_connection()->get_features();
+  uint64_t missing =
+    get_quorum_con_features() & ~m->get_connection()->get_features();
```
Comment (liewegas) [inline src/mon/Monitor.cc]: I don't understand why ``quorum_con_features`` is here.  We are probing, so there is no quorum--these features are only set in win_election and lose_election.

**Comment by smithfarm** [comment]
adding DNM since this is a cherry-pick of https://github.com/ceph/ceph/pull/28671 which has not yet landed in master

**Comment by smithfarm** [comment]
@jecluis Given that #28671 was closed and this is (possibly?) a cherry-pick of that PR, can we close this one now?

**Comment by smithfarm** [comment]
this is a cherry-pick of https://github.com/ceph/ceph/pull/28671 which has been closed, so closing this one as well

**Pull request closed without merging**